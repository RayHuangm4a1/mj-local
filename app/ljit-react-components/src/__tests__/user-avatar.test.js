import React from 'react';
import { shallow, mount, } from 'enzyme';
import UserAvatar from '../components/user-avatar';

const { SizeEnums, AlignmentEnums, } = UserAvatar;

describe('UserAvatar', () => {
	beforeEach(() => {
		jest.mock('antd/lib/avatar');
		jest.mock('../components/avatar', () => function mockComponent() {
			return <div />;
		});
	});

	afterEach(() => {
		jest.resetModules();
	});

	it('handle default props', () => {
		const {
			size,
			alignment,
			onError,
		} = UserAvatar.defaultProps;

		expect(size).toEqual(SizeEnums.SMALL);
		expect(alignment).toEqual(AlignmentEnums.LEFT);
		expect(onError).toBeInstanceOf(Function);
	});

	it('should render correctly', () => {
		const text = 'user';
		const wrapper = shallow(
			<UserAvatar
				text={text}
				size={56}
				alignment={AlignmentEnums.LEFT}
				userName='mock-username'
				description='mock-description'
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-user-avatar', () => {
		const wrapper = shallow(<UserAvatar />);

		expect(wrapper.hasClass('ljit-user-avatar')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<UserAvatar className={className} />);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	it('should be selectable by top: TOP', () => {
		const alignment = AlignmentEnums.TOP;
		const wrapper = shallow(<UserAvatar alignment={alignment} />);

		expect(wrapper.hasClass('ljit-user-avatar--top')).toEqual(true);
	});

	it('should be selectable by left: LEFT', () => {
		const alignment = AlignmentEnums.LEFT;
		const wrapper = shallow(<UserAvatar alignment={alignment} />);

		expect(wrapper.hasClass('ljit-user-avatar--left')).toEqual(true);
	});

	it('should mount in a full DOM', () => {
		const className = 'mock-class';
		const style = {};
		const src = 'https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png';
		const alt = 'ljit-avatar';
		const text = 'ljit';
		const userName = 'User001';
		const onError = () => {};
		const alignment = AlignmentEnums.TOP;
		const description = 'mock-description';
		const size = 56;
		const wrapper = mount(
			<UserAvatar
				className={className}
				style={style}
				src={src}
				alt={alt}
				text={text}
				userName={userName}
				onError={onError}
				alignment={alignment}
				description={description}
				size={size}
			/>
		);

		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().style).toMatchObject(style);
		expect(wrapper.props().src).toBe(src);
		expect(wrapper.props().alt).toBe(alt);
		expect(wrapper.props().text).toBe(text);
		expect(wrapper.props().userName).toBe(userName);
		expect(wrapper.props().onError).toEqual(onError);
		expect(wrapper.props().alignment).toBe(alignment);
		expect(wrapper.props().description).toEqual(description);
		expect(wrapper.props().size).toEqual(size);
	});

	describe('Alignment Enums', () => {
		it('should contains TOP property', () => {
			const typeName = 'top';
			const formatType = 'TOP';

			expect(AlignmentEnums[formatType]).toEqual(typeName);
		});
		it('should contains LEFT property', () => {
			const typeName = 'left';
			const formatType = 'LEFT';

			expect(AlignmentEnums[formatType]).toEqual(typeName);
		});
	});
});
