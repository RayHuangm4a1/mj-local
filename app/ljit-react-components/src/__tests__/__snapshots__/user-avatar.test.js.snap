// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserAvatar
    alignment="left"
    description="mock-description"
    onError={[Function]}
    size={56}
    text="user"
    userName="mock-username"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ljit-user-avatar__avatar"
        >
          <Avatar
            onError={[Function]}
            size={56}
            text="user"
          />
        </div>,
        <div
          className="ljit-user-avatar__content"
        >
          <div
            className="ljit-user-avatar__username"
          >
            mock-username
          </div>
          <div
            className="ljit-user-avatar__description"
          >
            mock-description
          </div>
        </div>,
      ],
      "className": "ljit-user-avatar ljit-user-avatar--left",
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Avatar
            onError={[Function]}
            size={56}
            text="user"
          />,
          "className": "ljit-user-avatar__avatar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": undefined,
            "onError": [Function],
            "size": 56,
            "src": undefined,
            "text": "user",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="ljit-user-avatar__username"
            >
              mock-username
            </div>,
            <div
              className="ljit-user-avatar__description"
            >
              mock-description
            </div>,
          ],
          "className": "ljit-user-avatar__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "mock-username",
              "className": "ljit-user-avatar__username",
            },
            "ref": null,
            "rendered": "mock-username",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "mock-description",
              "className": "ljit-user-avatar__description",
            },
            "ref": null,
            "rendered": "mock-description",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ljit-user-avatar__avatar"
          >
            <Avatar
              onError={[Function]}
              size={56}
              text="user"
            />
          </div>,
          <div
            className="ljit-user-avatar__content"
          >
            <div
              className="ljit-user-avatar__username"
            >
              mock-username
            </div>
            <div
              className="ljit-user-avatar__description"
            >
              mock-description
            </div>
          </div>,
        ],
        "className": "ljit-user-avatar ljit-user-avatar--left",
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Avatar
              onError={[Function]}
              size={56}
              text="user"
            />,
            "className": "ljit-user-avatar__avatar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": undefined,
              "onError": [Function],
              "size": 56,
              "src": undefined,
              "text": "user",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="ljit-user-avatar__username"
              >
                mock-username
              </div>,
              <div
                className="ljit-user-avatar__description"
              >
                mock-description
              </div>,
            ],
            "className": "ljit-user-avatar__content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "mock-username",
                "className": "ljit-user-avatar__username",
              },
              "ref": null,
              "rendered": "mock-username",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "mock-description",
                "className": "ljit-user-avatar__description",
              },
              "ref": null,
              "rendered": "mock-description",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
